Source/ACE.DatLoader/Entity/AnimationHooks/AttackHook.cs

```diff
@@ -0,0 +1,25 @@
+﻿namespace ACE.DatLoader.Entity.AnimationHooks
+{
+    public class AttackHook : IHook
+    {
+        public AttackCone AttackCone { get; set; }
+
+        public static AttackHook ReadHookType(DatReader datReader)
+        {
+            AttackHook a = new AttackHook();
+            AttackCone ac = new AttackCone();
+
+            ac.PartIndex = datReader.ReadUInt32();
```

Comment:
Consider giving AttackCode a "Read(DatReader datReader)" method, and then making AttackCone's property setters private.